gwpy/io/ligolw.py

```diff
@@ -533,10 +533,16 @@ def write_tables(target, tables, append=False, overwrite=False, **kwargs):
 
     # write file
     if isinstance(target, str):
-        kwargs.setdefault('gz', target.endswith('.gz'))
+        kwargs.setdefault(
+            'compress',
+            'gz' if target.endswith('.gz') else False,
+        )
         ligolw_utils.write_filename(xmldoc, target, **kwargs)
     elif isinstance(target, FILE_LIKE):
-        kwargs.setdefault('gz', target.name.endswith('.gz'))
+        kwargs.setdefault(
+            'compress',
+            'gz' if target.name.endswith('.gz') else False,
+        )
         ligolw_utils.write_fileobj(xmldoc, target, **kwargs)
```

Comment:
To support both I think I would rewrite this as

```python
    # find writer function and target filename
    if isinstance(target, FILE_LIKE):
        writer = ligolw_utils.write_fileobj
        name = target.name
    else:
        writer = ligolw_utils.write_filename
        name = str(target)

    # handle gzip compression kwargs
    if name.endswith('.gz') and writer.__module__.startswith('glue'):
        kwargs.setdefault('gz', True)
    elif name.endswith('.gz'):
        kwargs.setdefault('compress', 'gz')

    # write XML
    writer(xmldoc, target, **kwargs)

Reply:
Good idea @duncanmmacleod, I've done this in the latest commit